modules/xml/filterx-parse-windows-eventlog-xml.c

```diff
@@ -0,0 +1,277 @@
+/*
+ * Copyright (c) 2024 Axoflow
+ * Copyright (c) 2024 Attila Szakacs <attila.szakacs@axoflow.com>
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 as published
+ * by the Free Software Foundation, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+#include "filterx-parse-windows-eventlog-xml.h"
+#include "filterx/object-string.h"
+#include "filterx/object-dict-interface.h"
+#include "scratch-buffers.h"
+
+static void _set_error(GError **error, const gchar *format, ...) G_GNUC_PRINTF(2, 0);
+static void
+_set_error(GError **error, const gchar *format, ...)
+{
+  if (!error)
+    return;
+
+  va_list va;
+  va_start(va, format);
+  *error = g_error_new_valist(g_quark_from_static_string("filterx-parse-windows-eventlog-xml"), 0, format, va);
+  va_end(va);
+}
+
+
+typedef struct FilterXParseWindowsEventlogXmlUserData_
+{
+  FilterXParseXmlUserData super;
+  gboolean in_event, in_event_data, in_data, has_named_data;
+  GString *last_data_name;
+} FilterXParseWindowsEventlogXmlUserData;
```

Comment:
Just to be constructive as well, FilterXParseXMLState could be a name for the parent and
FilterXParseWEVTState could be the other, and similarly name structs for the functions.

Reply:
Yes, you understand it right. There are things that is needed for `parse_xml()`, but `parse_windows_eventlog_xml()` needs some other fields on top of that.

The thing is that this user data is ugly by nature. You will see 4 different types of user_data passing/casting in the code. I did not want to have another user_data in `FilterXParseXmlUserData` for the descendant class to fill, because casting turned out to be a hell, so I did this object oriented approach, which seemed better.

I can rename it, its not a problem. I can abbreviate the `WindowsEventlog` parts in the new code with `WinEvent`.

Reply:
I have just seen your suggestions, I will rename accordingly.

Reply:
Fixed.